main.go

```diff
@@ -80,212 +76,235 @@ func main() {
+		}
+		glog.Infof("Created network endpoint group %q", negName)
+
+		hcName := tagConfig.HealthCheck.Name
+		glog.Infof("Creating health check %q...", hcName)
+		if err := client.CreateHealthCheck(hcName, tagConfig.HealthCheck.Path); err != nil {
+			return fmt.Errorf("Can't create health check %q: %+v", hcName, err)
+		}
+		glog.Infof("Created health check %q", hcName)
+
+		bsName := tagConfig.BackendService.Name
+		glog.Infof("Creating backend service %q...", bsName)
+		if err := client.CreateBackendService(bsName, negName, hcName, tagConfig.BackendService.Affinity, tagConfig.BackendService.CDN); err != nil {
+			return fmt.Errorf("Can't create backend service %q: %+v", bsName, err)
+		}
+		glog.Infof("Created backend service %q", bsName)
+
+		glog.Infof("Updating URL map %q...", urlMapName)
+		if err := client.UpdateURLMap(urlMapName, bsName, tagConfig.BackendService.Host, tagConfig.BackendService.Path); err != nil {
+			return fmt.Errorf("Can't update URL map %q: %+v", urlMapName, err)
+		}
+		glog.Infof("Updated URL map %q", urlMapName)
 	}
-	return n
+
+	glog.Infof("Watching service with tag %q...", tagName)
+
+	return nil
+}
+
+func handleServiceDeletedUpdate(
+	client cloud.Cloud,
+	tagConfig *tagConfiguration,
+	lock *sync.RWMutex,
+	instances map[string]*registry.ServiceInstance,
+) error {
+	lock.Lock()
+	defer lock.Unlock()
+
+	var toRemove []gce.NetworkEndpoint
+
+	for k, instance := range instances {
+		toRemove = append(toRemove, gce.NetworkEndpoint{
+			Instance: normalizeInstanceName(instance.Host),
+			IP:       instance.Address,
+			Port:     instance.Port,
+		})
+		delete(instances, k)
+	}
+
+	//do we have instances to remove from the NEG?
+	if len(toRemove) > 0 {
+		negName := tagConfig.NetworkEndpointGroup.Name
+		if err := client.DetachEndpointsFromGroup(toRemove, negName); err != nil {
+			return fmt.Errorf("Failed detaching endpoints from %q network endpoint group: %+v", negName, err)
+		}
+	}
+
+	return nil
+}
+
+func handleServiceChangedUpdate(
+	client cloud.Cloud,
+	tagConfig *tagConfiguration,
+	lock *sync.RWMutex,
+	instances map[string]*registry.ServiceInstance,
+	update *registry.ServiceUpdate,
+) error {
+	lock.Lock()
+	defer lock.Unlock()
+
+	var toAdd, toRemove []gce.NetworkEndpoint
+
+	// finding instances to remove
+	for k, instance := range instances {
+		// Doesn't instance exist in update?
+		if _, ok := update.ServiceInstances[k]; !ok {
+			toRemove = append(toRemove, gce.NetworkEndpoint{
+				Instance: normalizeInstanceName(instance.Host),
+				IP:       instance.Address,
+				Port:     instance.Port,
+			})
+			delete(instances, k)
+		}
+	}
+
+	// finding instances to add
+	for k, instance := range update.ServiceInstances {
+		// Doesn't instance exist in running instances?
+		if _, ok := instances[k]; !ok {
+			toAdd = append(toAdd, gce.NetworkEndpoint{
+				Instance: normalizeInstanceName(instance.Host),
+				IP:       instance.Address,
+				Port:     instance.Port,
+			})
+			instances[k] = instance
```

Comment:
Ah, now I see where we forked this from. I have to be straightforward, that implementation is not great and it only has 10 stars, it is not a lot for implementation to be seen as stable or adopted by community.

In that sense we either need to fully adopt it within ourselves or ditch it as a risk going forward.

Reply:
If it's stable and adopted by community then no need to fork and change it 😉